stan/math/rev/core/operator_unary_negative.hpp

```diff
@@ -48,7 +49,30 @@ class neg_vari final : public op_v_vari {
  * @param a Argument variable.
  * @return Negation of variable.
  */
-inline var operator-(const var& a) { return {new internal::neg_vari(a.vi_)}; }
+inline var operator-(const var& a) {
+  return make_callback_var(-a.val(), [a](const auto vi) {
+    if (unlikely(is_nan(a.val()))) {
+      a.adj() = NOT_A_NUMBER;
```

Comment:
Did this cause a problem somewhere? I'm not a fan of these manual NaN injections. If `a` is NaN and it gets used anywhere, the rest of the calculation will probably blow up. I don't think we need to spend the `if` statements making sure that happens.

Reply:
We just did it in the old impl so I think we should do it in this impl as well. Unless we don't check when values are NaNs. It might be a discussion to have at a later time